R/util.R

```diff
@@ -130,27 +130,62 @@ util.is_vector_of_numbers <- function(x){
     # (regardless of whether x is numeric)
     # Character vectors sometimes need to be changed to numerics,
     # e.g., when all columns set to character types by default.
-
+  
+    # determine if x is logical first, because as.numeric will coerce logicals
+    # to boolean 0,1 rather than to NA, meaning util.is_vector_of_numbers would return TRUE
+    if(is.logical(x)) return(FALSE)
+    
     # find numeric values
-    numeric_values <- grepl("^-*[[:digit:]]*\\.*[[:digit:]]+$",x)
+    x_as_numeric <- suppressWarnings(as.numeric(x))
 
-    # find the blank values
-    blank_values <- util.is_blank(x)
+    # anything that gets coerced to NA by as.numeric is not really a number.
+    # but elements that were blank originally are ok.
+    blanks <- is.na(x_as_numeric)
```

Comment:
nit: based on your comments, I'm finding the use of `blank` in the variable names confusing. What do you think of this (logic should be identical):

```
non_numeric <- is.na(x_as_numeric)
originally_blank <- util.is_blank(x)
if(any(non_numeric & !originally_blank)) {
  ...
}
```